libs/server/Resp/RespCommandInfoParser.cs

```diff
@@ -0,0 +1,488 @@
+                else if (string.Equals(keySpecType, "keynum", StringComparison.Ordinal))
+                    parser = FindKeysKeyNumParser.Instance;
+                else if (string.Equals(keySpecType, "unknown", StringComparison.Ordinal))
+                    parser = FindKeysUnknownParser.Instance;
+                else return false;
+            }
+            else return false;
+
+            if (!parser.TryReadFromResp(ref ptr, end, out keySpecMethod)) return false;
+
+            return true;
+        }
+
+        /// <summary>
+        /// Tries to parse key spec header from RESP format
+        /// </summary>
+        /// <param name="ptr">Pointer to current RESP chunk to read</param>
+        /// <param name="end">Pointer to end of RESP chunk to read</param>
+        /// <param name="keySpecType">Parsed key spec type</param>
+        /// <returns>True if parsing successful</returns>
+        private static unsafe bool TryReadKeySpecHeader(ref byte* ptr, byte* end, out string keySpecType)
+        {
+            keySpecType = default;
+
+            RespReadUtils.ReadArrayLength(out var ksTypeElemCount, ref ptr, end);
+            if (ksTypeElemCount != 4) return false;
+            RespReadUtils.ReadStringWithLengthHeader(out var ksTypeStr, ref ptr, end);
+            if (!string.Equals(ksTypeStr, "type", StringComparison.Ordinal)) return false;
+            RespReadUtils.ReadStringWithLengthHeader(out var ksType, ref ptr, end);
+            RespReadUtils.ReadStringWithLengthHeader(out var ksSpecStr, ref ptr, end);
+            if (!string.Equals(ksSpecStr, "spec", StringComparison.Ordinal)) return false;
+
+            keySpecType = ksType;
+            return true;
+        }
+
+        /// <summary>
+        /// Interface for classes implementing parsing of KeySpecMethodBase objects
+        /// </summary>
+        internal interface IKeySpecParser
+        {
+            /// <summary>
+            /// Tries to parse KeySpecMethodBase from RESP format
+            /// </summary>
+            /// <param name="ptr">Pointer to current RESP chunk to read</param>
+            /// <param name="end">Pointer to end of RESP chunk to read</param>
+            /// <param name="keySpecMethod">Parsed KeySpecMethodBase object</param>
+            /// <returns>True if parsing successful</returns>
+            unsafe bool TryReadFromResp(ref byte* ptr, byte* end, out KeySpecMethodBase keySpecMethod);
+        }
+
+        /// <summary>
+        /// Parser for the BeginSearchIndex key specification method
+        /// </summary>
+        internal sealed class BeginSearchIndexParser : IKeySpecParser
+        {
+            private static BeginSearchIndexParser instance;
+
+            private BeginSearchIndexParser()
+            {
+            }
+
+            public static BeginSearchIndexParser Instance
+            {
+                get { return instance ??= new BeginSearchIndexParser(); }
+            }
```

Comment:
I would personally comment this as follows:

```suggestion
            /// <summary>
            /// Disallow default constructor (singleton)
            // </summary>
            private BeginSearchIndexParser() { }

            /// <summary>
            /// Returns the singleton instance of <see cref="BeginSearchIndexParser">.
            /// </summary>
            public static BeginSearchIndexParser Instance
            {
                get { return instance ??= new BeginSearchIndexParser(); }
            }
```

And obviously the same for the remaining singletons.